### DISCUSSION TITLE
UX: compress files and folders and pipe to age under Windows

### DISCUSSION THREAD

**Discussion by graphixillusion** [FiloSottile/age#248] [Q&A] [answered]
Hi guys. I'm experimenting some CLI commands to make compression, archiving and encryption on the fly using Windows cmd. I successfully managed to achieve the compression and encryption of one single file using this command:
**```7z a dummy -tgzip -so filename_to_compress | age -r publickey > encrypted.file ```**
Now i'm trying to achieve the same thing with Folders (with multiple files inside) but i'm not lucky this time. I see that **7z -so** switch works just with this formats: **xz, gzip, bzip2 and tar** which work just with singles files. I'm just wondering what can i use and how to make it works under Windows, using 7z or maybe something else: any idea? 
I have another quick question to ask: is it possible to pipe and use age -p switch (with auto generated passphrase?)

Thank you all!

**Comment by graphixillusion** [comment, marked as answer]
Actually i have found that this command works good:

**```7z a dummy -ttar -so folder_to_compress | age -r publickey > encrypted.file```**

What about the -p switch support with this command? Thank you.

**Reply by FiloSottile** [reply]
Using `-p` and standard input at the same time on Windows is #196.
